pkg/neonvm/cpuscaling/sysfs.go

```diff
@@ -0,0 +1,127 @@
+package cpuscaling
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+	"strconv"
+	"strings"
+)
+
+// CPU directory path
+const cpuPath = "/sys/devices/system/cpu/"
+
+type cpuSysfsState struct{}
+
+func (cs *cpuSysfsState) SetState(cpuNum int, cpuState cpuState) error {
+	var state string
+	switch cpuState {
+	case cpuOnline:
+		state = "1"
+	case cpuOffline:
+		state = "0"
+	}
+
+	err := os.WriteFile(filepath.Join(cpuPath, fmt.Sprintf("cpu%d/online", cpuNum)), []byte(state), 0o644)
+	if err != nil {
+		return fmt.Errorf("failed to set CPU %d online status: %w", cpuNum, err)
+	}
+
+	return nil
+}
+
+func (cs *cpuSysfsState) GetState(cpuNum int) (cpuState, error) {
+	data, err := os.ReadFile(filepath.Join(cpuPath, "online"))
+	if err != nil {
+		return cpuOffline, err
+	}
+
+	onlineCPUs, err := cs.parseMultipleCPURange(string(data))
+	if err != nil {
+		return cpuOffline, err
+	}
+
+	if slices.Contains(onlineCPUs, cpuNum) {
+		return cpuOnline, nil
+	}
+
+	return cpuOffline, nil
+}
+
+// PossibleCPUs returns the start and end indexes of all possible CPUs.
+func (cs *cpuSysfsState) PossibleCPUs() (int, int, error) {
+	data, err := os.ReadFile(filepath.Join(cpuPath, "possible"))
+	if err != nil {
+		return -1, -1, err
+	}
+
+	return cs.parseCPURange(string(data))
+}
+
+// parseCPURange parses the CPU range string (e.g., "0-3") and returns a list of CPUs.
+func (cs *cpuSysfsState) parseCPURange(cpuRange string) (int, int, error) {
+	cpuRange = strings.TrimSpace(cpuRange)
+	parts := strings.Split(cpuRange, "-")
+
+	// Single CPU case, e.g., "0"
+	if len(parts) == 1 {
+		cpu, err := strconv.Atoi(parts[0])
+		if err != nil {
+			return -1, -1, err
+		}
+		return cpu, cpu, nil
+	}
+
+	// Range case, e.g., "0-3"
+	start, err := strconv.Atoi(parts[0])
+	if err != nil {
+		return -1, -1, err
+	}
+	end, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return -1, -1, err
+	}
+	return start, end, nil
+}
+
+// parseMultipleCPURange parses the multiple CPU range string (e.g., "0-3,5-7") and returns a list of CPUs.
+func (cs *cpuSysfsState) parseMultipleCPURange(cpuRange string) ([]int, error) {
```

Comment:
```suggestion
func (cs *cpuSysfsState) parseMultipleCPURange(cpuRanges string) ([]int, error) {
```

(because it's not just one range ?)